kevin218/Eureka
# BIG TESTING UPDATE: Read for more info

Hello everybody! In this pull request we have a test for NIRCam up and running. These tests are intended for anyone who is uploading new code to run them and verify that none of their changes have broken the basic functionality of the code (a common example: changes in function names, etc). **It is recommended everyone runs this test whenever they upload new code**. The reason behind this is to make the workflow between collaborators easier, and to be thoughtful of everyone who will use your new code so they don't have to track down small changes and bugs. 

**HOW TO RUN TESTS**
----------------------------------

1. Make sure you're in the eureka/tests folder
2. run 'pytest'
3. If you wish to only run the NIRCam test: pytest -k test_NIRCam

**What about NIRSpec and MIRI?**
-----------------------------------------------
The full code functionality of NIRSpec and MIRI runs is still being tested. The need for smaller files for test runs for NIRSpec and MIRI is still a task at hand, and issue #86 was opened pertaining errors with NIRSpec data, therefore there isn't any working NIRSpec and MIRI tests yet, despite being written into the test file (commented out). But @jbrande will be helping with that.

**RUNDOWN OF CHANGES IN THIS PR**
------------------------------------------------------

- `modified:   ../../demos/S3/S3_template.ecf` - added 'save_output' meta parameter in order to allow tests to still save outputs necessary for also running S4 tests. Controls whether S3 outputs will be saved into files
- `modified:   ../S3_data_reduction/s3_reduce.py` - inlcudes save_output option in order to allow tests to still save S3 outputs (see above)
- `modified:   ../S4_generate_lightcurves/s4_genLC.py` - included aperture values for S4 output filenames (such that it matches the naming case from S3) *see lines 146,147*
- `modified:   ../lib/util.py` - updated description of path function from 'create path' to 'reads path'
- `modified:   test_general.py` - the meat of this PR: where all the new tests are written
